content/blog/2022-06-14-DVC-VS-Code-extension.md

```diff
@@ -0,0 +1,158 @@
+
+# Why an extension?
+
+We built DVC to expand upon the Git workflow. This approach has made it a
+powerful tool that can easily be picked up by users familiar with Git. However,
+it can be more difficult for those less familiar with CLI tools to get
+accustomed to using DVC. Moreover, while DVC provides many useful
+visualizations, a CLI tool can only take things so far.
+
+Therefore we wanted to incorporate DVC's functionality in a graphical interface.
+[VS Code is the IDE of choice for many](https://insights.stackoverflow.com/survey/2021#section-most-popular-technologies-integrated-development-environment),
+including most of us here at Iterative. It seemed like the natural choice for a
+platform to make DVC more versatile, accessible, and useful.
+
+With this extension, we want to achieve the following:
+
+- Move the entire data science workflow into your IDE
+- Enhance existing visualizations and provide new ones
+- Make DVC more accessible by hiding the complexity of the CLI
+
+As data scientists, DVC is our toolbox. This extension turns VS Code into our
+workshop.
+
+<!-- # Why Visual Studio Code
+
+One of [our core beliefs at Iterative](https://iterative.ai/why-iterative/) is
+that our products should couple tightly with existing tools and processes for
+data science projects.
+[VS Code is the IDE of choice for many](https://insights.stackoverflow.com/survey/2021#section-most-popular-technologies-integrated-development-environment),
+including many of us at Iterative. Here are a few reasons why we really like it:
+
+VS Code...
+
+- is open source and lightweight
+- has excellent support for Git workflows
+- supports Jupyter Notebooks
+- comes with remote development and execution built-in
+- has a rich ecosystem of extensions -->
+
+# Features
+
+Our first priority was to include all DVC features related to data management
+and experiments. Here are the main features that are available when you install
+the extension right now:
+
+## DVC View: experiments, metrics, and plots
+
+The main feature of our extension is the DVC view, which provides a one-stop
+shop for everything related to your machine learning experiments. You can run
+new experiments from here and compare both metrics and plots for different
+models. Never has it been easier to analyze the results of your experiments and
+find the best model.
+
+![Experiment tables to compare different
+experiments](/uploads/images/2022-06-14/dvc-view-experiments-table.png)
+
+![Plots to visualize your
+experiments](/uploads/images/2022-06-14/dvc-view-plots.png)
+
+## Source control management
+
+On top of Git version contol for your code, the source control view now lets you
+manage datasets and models tracked by DVC. Quickly see which artifacts have been
+changed, and synchronize versions with your remote repository. Use `checkout`,
+`commit`, `add`, `push`, and `pull` straight from the interface:
```

Comment:
Hmm, I see what you mean. On the other hand, if people using VSCode know anything about CLI, it might just be Git.

I can remove this sentence altogether or keep it here. Functionality-wise, I think it's already described in the two preceding sentences. Thoughts?

Reply:
I didn't even think about Git but you're right, they may be aware of `git checkout/commit/etc`... which is even more confusing since this paragraph is more about DVC 😋 and b/c you can already `git checkout/etc` from the Source Control view. I'd just bring it one level up in abstraction, like this:

> ... the Source Control view now lets you version your data and model files with DVC (without using the terminal. It also shows you the state of the workspace, similar to the familiar Git panel.

Idk, something like that.

Reply:
Thanks, updated!